src/lib/components/BaseModalElement/index.js

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseElement} from "../BaseElement";
+import "wicg-inert";
+import {handleOverflow} from "../../utils/handle-overflow";
+import {openModal} from "../../actions";
+import {closeModal} from "../../actions";
+import "./_styles.scss";
+
+/**
+ * Base element that provides modal functionality.
+ * Handles inert state for document, focus management, and modal content overflow.
+ *
+ * @extends {BaseElement}
+ */
+export class BaseModalElement extends BaseElement {
+  static get properties() {
+    return {
+      open: {type: Boolean, reflect: true},
+      animatable: {type: Boolean, reflect: true},
+      overflow: {type: Boolean, reflect: true},
+      parentModal: {attribute: "parent-modal", reflect: true},
+    };
+  }
+
+  constructor() {
+    super();
+
+    this.open_ = false;
+    this.animatable = false;
+    this.inert = true;
+    this.overflow = false;
+    this.idSalt = BaseElement.generateIdSalt("web-modal-");
+
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.onResize = this.onResize.bind(this);
+    this.onAnimationEnd = this.onAnimationEnd.bind(this);
+    this.manageDocument = this.manageDocument.bind(this);
+    this.manageFocus = this.manageFocus.bind(this);
+    this.getTrigger = this.getTrigger.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener("click", this.onClick);
+    // Set tabindex to -1 so modal can be focused when it's opened.
+    this.tabIndex = -1;
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener("click", this.onClick);
+  }
+
+  set open(val) {
+    if (this.open_ === val) {
+      return;
+    }
+
+    const oldVal = this.open_;
+
+    this.open_ = val;
+    if (this.open_) {
+      // Must get trigger before manipulating the DOM.
+      this.getTrigger();
```

Comment:
Just save `this._triggerElement = document.activeElement`. Adding a class is needless when you can just take a ref. (Yes, that element may no longer be in the DOM when the page changes. But you unref it then.)